backend/src/components/secureExchange.js

```diff
@@ -556,45 +556,80 @@ async function removeSecureExchangeStudent(req, res) {
+    return res.status(HttpStatus.OK).json(result);
+  } catch (e) {
+    log.error(e, 'updateEdxUserRoles', 'Error occurred while attempting to update user roles.');
+    return handleExceptionResponse(e, res);
+  }
+}
+
+async function updateEdxUserDistrictRoles(req, res) {
+  try {
+    const token = getAccessToken(req);
+    validateAccessToken(token);
+    checkEDXUserDistrictAdminPermission(req);
+    checkEDXUserAccess(req, 'DISTRICT', req.body.params.districtId);
 
-    //Roles to be added
+    let edxUser = await getData(token, `${config.get('edx:edxUsersURL')}/${req.body.params.edxUserID}`, req.session?.correlationID);
+    let selectedUserDistricts = edxUser.edxUserDistricts.filter(district => district.districtID === req.body.params.districtId);
+    if (!selectedUserDistricts[0]) {
+      return errorResponse(res, 'A user district entry was not found for the selected user.', HttpStatus.NOT_FOUND);
+    }
+    if (selectedUserDistricts.length > 1) {
+      return errorResponse(res, 'Too many user district entries have been found for the selected user.', HttpStatus.CONFLICT);
+    }
+    let selectedUserDistrict = selectedUserDistricts[0];
+    let existingUserDistrictRoles = new Map(selectedUserDistrict.edxUserDistrictRoles.map(edxUserDistrictRole => [edxUserDistrictRole.edxRoleCode, edxUserDistrictRole]));
+
+    selectedUserDistrict.edxUserDistrictRoles = [];
     req.body.params.selectedRoles.forEach(function (role) {
-      if (!selectedUserSchool[0].edxUserSchoolRoles.filter(value => role === value.edxRoleCode).length > 0) {
-        let newRole = {};
-        newRole.edxUserSchoolID = selectedUserSchool[0].edxUserSchoolID;
-        newRole.edxRoleCode = role;
-        selectedUserSchool[0].edxUserSchoolRoles.push(newRole);
+      if (existingUserDistrictRoles.has(role)) {
+        selectedUserDistrict.edxUserDistrictRoles.push(existingUserDistrictRoles.get(role));
+        return;
       }
+      let newRole = {};
+      newRole.edxUserDistrictID = selectedUserDistrict.edxUserDistrictID;
+      newRole.edxRoleCode = role;
+      selectedUserDistrict.edxUserDistrictRoles.push(newRole);
     });
 
-    selectedUserSchool[0].updateDate = null;
-    selectedUserSchool[0].createDate = null;
-
-    const payload = {
-      ...selectedUserSchool[0]
-    };
+    selectedUserDistrict.updateDate = null;
+    selectedUserDistrict.createDate = null;
 
-    const result = await putData(token, payload, config.get('edx:edxUsersURL') + '/' + selectedUserSchool[0].edxUserID + '/school', req.session?.correlationID);
+    const result = await putData(token, selectedUserDistrict, `${config.get('edx:edxUsersURL')}/${selectedUserDistrict.edxUserID}/district`, req.session?.correlationID);
     return res.status(HttpStatus.OK).json(result);
   } catch (e) {
     log.error(e, 'updateEdxUserRoles', 'Error occurred while attempting to update user roles.');
```

Comment:
I think this should be updateEdxUserDistrictRoles ... update user district roles.